packages/components/src/__future__/Tag/RemoveButton.module.scss

```diff
@@ -0,0 +1,77 @@
+@import "~@kaizen/design-tokens/sass/border";
+@import "~@kaizen/design-tokens/sass/color";
+@import "../../../styles/button-reset";
+
+.removeButton {
+  @include button-reset;
+
+  position: relative;
+  display: inline-flex;
+  border-radius: 50%;
+
+  &:focus {
+    outline: none;
+  }
+
+  &:focus-visible {
+    &::after {
+      $focus-ring-offset: -3px;
+
+      border-color: $color-blue-500;
+      content: "";
+      position: absolute;
+      background: transparent;
+      border-width: $border-focus-ring-border-width;
+      border-style: $border-focus-ring-border-style;
+      border-radius: 50%;
+      inset: $focus-ring-offset;
+    }
+  }
+
+  &::before {
+    position: absolute;
+    content: "";
+    inset: -5px;
+    border-radius: 50%;
+  }
+
+  .closeIcon {
+    fill: rgba($color-purple-800-rgb, 0.7);
+  }
+
+  &:hover {
+    cursor: pointer;
+
+    .closeIcon {
+      fill: $color-purple-800;
+    }
+
+    &.gray {
+      background-color: rgba($color-gray-600-rgb, 0.2);
+    }
+
+    &.blue {
+      background-color: rgba($color-blue-500-rgb, 0.3);
+    }
+
+    &.green {
+      background-color: rgba($color-green-500-rgb, 0.3);
+    }
+
+    &.yellow {
+      background-color: rgba($color-yellow-500-rgb, 0.5);
+    }
+
+    &.orange {
+      background-color: rgba($color-orange-500-rgb, 0.3);
+    }
+
+    &.red {
+      background-color: rgba($color-red-500-rgb, 0.3);
+    }
+
+    &.purple {
+      background-color: rgba($color-purple-500-rgb, 0.3);
+    }
+  }
```

Comment:
Pull all of these out so they aren't nested

Reply:
This seems fine to me, a matter of personal preference. Grouping by hover state isn't uncommon or harmful.